### PR TITLE
WIP - Fix custom DataBus documentation for Nsb v6

### PR DESCRIPTION
**Pull request by indualagarsamy**
The documentation for custom DataBus does not address any of the changes needed for Version 6. Users had to spelunk other custom databus implementations to get this working.

### CODE REVIEW

**Comment by indualagarsamy** [comment]
@mikeminutillo @SimonCropp - Do you see value in showing how to extend the DatabusDefinition to allow other properties? For example, let's say for a Redis implementation a connection string is required. Thoughts? Or do you think it might be better implemented as a dedicated sample? I can create one.

File: nservicebus/messaging/databus/custom-databus_content_core_[6.0,7.0].partial.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
+It is possible to create a custom DataBus implementation. This is done by making use of the [Features](/nservicebus/pipeline/features.md) extension.
+
+
+## Implement the `IDataBus` interface. 
+
+This new class will provide the custom implementations for the `Get` and `Put` methods for the DataBus. 
+
+snippet: CustomDataBus
+
+This new implementation needs to be now registered as a new feature.
```
Comment (mikeminutillo) [inline nservicebus/messaging/databus/custom-databus_content_core_[6.0,7.0].partial.md]: I'd remove the word now.

File: nservicebus/messaging/databus/custom-databus_content_core_[6.0,7.0].partial.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+This new class will provide the custom implementations for the `Get` and `Put` methods for the DataBus. 
+
+snippet: CustomDataBus
+
+This new implementation needs to be now registered as a new feature. 
+
+
+## Define a feature 
+
+Define a [new feature](/nservicebus/pipeline/features.md) for the custom databus by creating a class that inherits from `Feature`. In the Setup method, register the custom DataBus implementation class.
```
Comment (mikeminutillo) [inline nservicebus/messaging/databus/custom-databus_content_core_[6.0,7.0].partial.md]: I would simplify this to just

> Define a new feature that registers the custom DataBus implementation class. 

Try not to describe the details of the snippet. That way, if the snippet changes, the text remains accurate.

File: nservicebus/messaging/databus/custom-databus_content_core_[6.0,7.0].partial.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+snippet: CustomDataBusFeature
+
+
+## Define a DataBusDefinition
+
+Define a new class which inherits from the `DataBusDefinition` class.
```
Comment (mikeminutillo) [inline nservicebus/messaging/databus/custom-databus_content_core_[6.0,7.0].partial.md]: Why is this needed? What does it do?

File: nservicebus/messaging/databus/custom-databus_content_core_[6.0,7.0].partial.md [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+## Define a DataBusDefinition
+
+Define a new class which inherits from the `DataBusDefinition` class. 
+
+snippet: CustomDataBusDefinition
+
+
+## Configure the endpoint
+
+Configure the endpoint to use the custom DataBus implementation instead of the default DataBus, as shown:
```
Comment (mikeminutillo) [inline nservicebus/messaging/databus/custom-databus_content_core_[6.0,7.0].partial.md]: Drop "as shown"

File: nservicebus/messaging/databus/custom.md [hunk 5]
```diff
@@ -12,10 +11,6 @@ redirects:
...
-It is possible to create a custom DataBus implementation by implementing the `IDataBus` interface, such as in the following minimalistic sample:
+partial: databus_content
 
-snippet: CustomDataBus
-
-To configure the endpoint to use the custom DataBus implementation it is enough to register it at endpoint configuration time, such as in the following sample:
-
-snippet: PluginCustomDataBus
\ No newline at end of file
+When customizing DataBus, it may be desirable to also [customize the DataBus serializer](/samples/databus/custom-serializer/) to use Json instead of binary serialization.
```
Comment (mikeminutillo) [inline nservicebus/messaging/databus/custom.md]: may be desirable? How would I make that decision? Should this just be a related article?

File: nservicebus/messaging/databus/custom-databus_content_core_[6.0,7.0].partial.md (hunk 3 above)
Comment (indualagarsamy) [inline nservicebus/messaging/databus/custom-databus_content_core_[6.0,7.0].partial.md]: This is the class that's used in the configuration of UseDataBus<> method/

File: nservicebus/messaging/databus/custom.md (hunk 5 above)
Comment (indualagarsamy) [inline nservicebus/messaging/databus/custom.md]: yeah, I think Related articles would be better.

**Pull request merged**